apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -304,21 +304,45 @@ SOMAExperimentAxisQuery <- R6::R6Class(
       # Construct the dimension names
       dim_names <- switch(collection,
         X = list(obs_labels, var_labels),
-        obsm = list(obs_labels, unique(tbl$soma_dim_1)$as_vector()),
-        varm = list(var_labels, unique(tbl$soma_dim_1)$as_vector()),
+        obsm = {
+          soma_dim_1 <- range(tbl$soma_dim_1$as_vector())
+          list(obs_labels, seq(min(soma_dim_1), max(soma_dim_1)))
+        },
+        varm = {
+          soma_dim_1 <- range(tbl$soma_dim_1$as_vector())
+          list(var_labels, seq(min(soma_dim_1), max(soma_dim_1)))
+        },
         obsp = list(obs_labels, obs_labels),
         varp = list(var_labels, var_labels)
       )
 
       # Use joinids if the dimension names are empty
       dim_names <- Map("%||%", dim_names, coords)
 
+      dims <- switch(
+        EXPR = collection,
+        X = list(
+          self$indexer$by_obs(coords$soma_dim_0),
+          self$indexer$by_var(coords$soma_dim_1)
+        ),
+        obsm = list(
+          self$indexer$by_obs(coords$soma_dim_0),
+          mat_coords$j
+        ),
+        varm = list(
+          self$indexer$by_var(coords$soma_dim_0),
+          mat_coords$j
+        ),
+        obsp = lapply(coords, self$indexer$by_obs),
+        varp = lapply(coords, self$indexer$by_var)
+      )
```

Comment:
I don't understand why we need to add this extra switch statement or change the way `dims` are calculated below. The tests all still passed after removing lines 322–338 and restoring `vapply_int(dim_names, length)`.  Maybe I'm missing something?

Reply:
Whoops, my bad. You're right, this is unnecessary and I've removed it